e2etest/local_transfer_test.go

```diff
@@ -312,28 +311,29 @@ func TestLocalTransfer(t *testing.T) {
 	}
 
 	// create server
-	ctx := context.Background()
 	svr, err := itx.NewServer(cfg)
 	require.NoError(err)
 
 	// Create and start probe server
+	ctx := context.Background()
 	probeSvr := probe.New(7788)
-	require.NoError(probeSvr.Start(context.Background()))
-
-	// Start server
-	go itx.StartServer(context.Background(), svr, probeSvr, cfg)
+	require.NoError(probeSvr.Start(ctx))
 	defer func() {
-		time.Sleep(10 * time.Second)
 		require.NoError(probeSvr.Stop(ctx))
-		require.NoError(svr.Stop(ctx))
 	}()
 
+	// Start server
+	ctx, cancel := context.WithTimeout(ctx, 32*time.Second)
+	defer cancel()
+	go itx.StartServer(ctx, svr, probeSvr, cfg)
```

Comment:
should use a timeout context, which calls svr.Stop() when done

currently, the svr is never stopped